src/platforms/java/common/performance/instrumentation/okhttp.mdx

```diff
@@ -0,0 +1,294 @@
+fun run(url: String): String? {
+  val request = Request.Builder()
+    .url(url)
+    .build()
+
+  val bodyStr = client
+    .newCall(request)
+    .execute()
+    .body?.toString()
+
+  Sentry.captureMessage("The Message $bodyStr")
+
+  return bodyStr
+}
+```
+
+```java
+import io.sentry.Sentry;
+import java.io.IOException;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.ResponseBody;
+
+String run(String url) throws IOException {
+  Request request = new Request.Builder()
+    .url(url)
+    .build();
+
+  ResponseBody body = client
+    .newCall(request)
+    .execute()
+    .body();
+
+  String bodyStr = body != null ? body.toString() : null;
+
+  Sentry.captureMessage("The Message " + bodyStr);
+
+  return bodyStr;
+}
+```
+
+To view and resolve the recorded message, log into [sentry.io](https://sentry.io) and open your project. Clicking on the error's title will open a page where you can see detailed information and mark it as resolved.
+
+## Customize the Recorded Span
+
+The captured span can be customized or dropped with a `BeforeSpanCallback`:
+
+```kotlin
+import io.sentry.ISpan
+import io.sentry.okhttp.SentryOkHttpInterceptor
+import okhttp3.Request
+import okhttp3.Response
+
+class CustomBeforeSpanCallback : SentryOkHttpInterceptor.BeforeSpanCallback {
+  override fun execute(span: ISpan, request: Request, response: Response?): ISpan? {
+    return if (request.url.toUri().toString().contains("/admin")) {
+      null
+    } else {
+      span
+    }
+  }
+}
+```
+
+```java
+import io.sentry.ISpan;
+import io.sentry.okhttp.SentryOkHttpInterceptor;
+import okhttp3.Request;
+import okhttp3.Response;
+
+class CustomBeforeSpanCallback implements SentryOkHttpInterceptor.BeforeSpanCallback {
+  public ISpan execute(ISpan span, Request request, Response response) {
+    if (request.getUrl().toUri().toString().contains("/admin")) {
+      return null;
+    } else {
+      return span;
+    }
+  }
+}
+```
+
+The callback instance must be set on the `SentryOkHttpInterceptor` once you create your `OkHttpClient` instance.
+
+```kotlin
+import okhttp3.OkHttpClient
+import io.sentry.okhttp.SentryOkHttpEventListener
+import io.sentry.okhttp.SentryOkHttpInterceptor
+
+private val client = OkHttpClient.Builder()
+  .eventListener(SentryOkHttpEventListener())
+  .addInterceptor(SentryOkHttpInterceptor(CustomBeforeSpanCallback()))
+  .build()
+```
+
+```java
+import okhttp3.OkHttpClient;
+import io.sentry.okhttp.SentryOkHttpEventListener;
+import io.sentry.okhttp.SentryOkHttpInterceptor;
+
+private final OkHttpClient client = new OkHttpClient.Builder()
+  .eventListener(new SentryOkHttpEventListener())
+  .addInterceptor(new SentryOkHttpInterceptor(new CustomBeforeSpanCallback()))
+  .build();
+```
+
+## Using Multiple Event Listeners
+
+The event listener can propagate calls to another `EventListener` or `EventListener.Factory` that's set on the `SentryOkHttpEventListener` once you create your `OkHttpClient` instance.
```

Comment:
Is this edit correct? Since this is the first time we reference an event listener on the page, I think it's better to specify.

```suggestion
The integration's event listener can propagate calls to another `EventListener` or `EventListener.Factory` that's set on the `SentryOkHttpEventListener` once you create your `OkHttpClient` instance.
```